futures-core/src/lib.rs

```diff
@@ -20,19 +22,33 @@ macro_rules! if_std {
     )*)
 }
 
-#[macro_use]
+#[macro_export]
+macro_rules! pinned_deref {
```

Comment:
Note that these functions aren't safe in general for arbitrary structures, especially with my proposed extension (which would block you from projecting from a `Pin<ManuallyDrop<T>>` to `Pin<T>`). Other examples include structures which have public fields but which also rely on `Pin` methods as a mechanism to ensure that those fields aren't mutated.

Reply:
Fixed.